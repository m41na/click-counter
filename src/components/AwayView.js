import React from 'react';

export default ({reset, value}) => {

    return (
        <section>
            <button type="button" onClick={reset}>reset</button>
            <br/>
            <span className="value">{value}</span>
        </section>
    );
}