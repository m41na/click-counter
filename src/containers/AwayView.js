import {connect} from 'react-redux';
import {resetAction} from '../actions';
import AwayView from '../components/AwayView';

const mapStateToProps = state => ({
    value: state.distance.value
});

const mapDispatchToProps = dispatch => ({
    reset: () => {
        dispatch(resetAction())
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(AwayView);