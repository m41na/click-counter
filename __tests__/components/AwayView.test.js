import React from 'react';
import { shallow } from 'enzyme';
import AwayView from '#/components/AwayView';

const initialProps = {
    value: 20,
    reset(){
        console.log('reset distance');
    }
};

describe('AwayView behavior', () => {
    describe('render()', () => {        
        it('should render with correct initial value', () => {
            const component = shallow(<AwayView {...initialProps} />);
            expect(component).toMatchSnapshot();        

            //assert span has initial value of 100
            const span = component.find("span.value");
            console.log('span value', span.text());
            expect(span.text()).toEqual('20');
        });
    });
});