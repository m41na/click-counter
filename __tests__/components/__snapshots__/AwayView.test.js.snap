// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwayView behavior render() should render with correct initial value 1`] = `
<section>
  <button
    onClick={[Function]}
    type="button"
  >
    reset
  </button>
  <br />
  <span
    className="value"
  >
    20
  </span>
</section>
`;
